woocommerce/woocommerce-android
# Fix navigation issue when triggering blaze from product detail

Closes: #12762 


### Description

This was way trickier than I expected considering how minor/simple it look the navigation bug. 

When triggering Blaze creation flow from product detail, the intro screen is always shown, event when the user has already several campaigns created. This PR fixes that.

I'm not super happy with the final solution so I'm open to hearing alternatives suggestions, but after trying different approaches this was the only one I found that worked for all cases. 

I'll try to explain the reason for the bug and the proposed solution. 

**Cause of the bug**
With the current implementation from `trunk` we use the property `graph` of the `NavController`; this property points to the main graph attached to the current NavHost, which is `nav_graph_main` (when the app is in single pane mode, more on this later), so when we call findNode on it, we retrieve the [nested graph](https://github.com/woocommerce/woocommerce-android/blob/2cbe776d5a120a79298cd1aebe59b7bbb3ae06d0/WooCommerce/src/main/res/navigation/nav_graph_main.xml#L18) attached to it, and not the [one](https://github.com/woocommerce/woocommerce-android/blob/2cbe776d5a120a79298cd1aebe59b7bbb3ae06d0/WooCommerce/src/main/res/navigation/nav_graph_products.xml#L10) we want to modify.

**Proposed solution**
- Retrieve the parent nav_graph instead and override the start destination there. 

### Testing information
1. From a Blaze eligible site with multiple campaigns created
2. Open product detail
3. Tap on `Promote with Blaze`
4. See how intro screen is never shown

Repeat steps 1-3 on a site eligible for Blaze but without any Blazd campaigns. Verify that intro screen is displayed. 

### The tests that have been performed

Smoked tested triggering Blaze campaign creation flow from all possible access points. 
I also verified that the events `🔵 Tracked: blaze_entry_point_tapped, Properties: {"source":"intro_view",` are still tracked correctly for each of the cases. 


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
